api/v1alpha4/conversion.go

```diff
@@ -16,12 +16,123 @@ limitations under the License.
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+//Constant variables
+const (
+	APIEndpointPort = "6443"
+)
+
+func (src *Metal3Cluster) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha5.Metal3Cluster)
+	return Convert_v1alpha4_Metal3Cluster_To_v1alpha5_Metal3Cluster(src, dst, nil)
+}
+
+func (dst *Metal3Cluster) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*v1alpha5.Metal3Cluster)
+	return Convert_v1alpha5_Metal3Cluster_To_v1alpha4_Metal3Cluster(src, dst, nil)
+}
+
+func (src *Metal3ClusterList) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha5.Metal3ClusterList)
+	return Convert_v1alpha4_Metal3ClusterList_To_v1alpha5_Metal3ClusterList(src, dst, nil)
+}
+
+func (dst *Metal3ClusterList) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*v1alpha5.Metal3ClusterList)
+	return Convert_v1alpha5_Metal3ClusterList_To_v1alpha4_Metal3ClusterList(src, dst, nil)
+}
+
+func (src *Metal3Machine) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha5.Metal3Machine)
+	if err := Convert_v1alpha4_Metal3Machine_To_v1alpha5_Metal3Machine(src, dst, nil); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (dst *Metal3Machine) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*v1alpha5.Metal3Machine)
+	if err := Convert_v1alpha5_Metal3Machine_To_v1alpha4_Metal3Machine(src, dst, nil); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (src *Metal3MachineList) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha5.Metal3MachineList)
+	return Convert_v1alpha4_Metal3MachineList_To_v1alpha5_Metal3MachineList(src, dst, nil)
+}
+
+func (dst *Metal3MachineList) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*v1alpha5.Metal3MachineList)
+	return Convert_v1alpha5_Metal3MachineList_To_v1alpha4_Metal3MachineList(src, dst, nil)
+}
+
+func (src *Metal3MachineTemplate) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha5.Metal3MachineTemplate)
+	if err := Convert_v1alpha4_Metal3MachineTemplate_To_v1alpha5_Metal3MachineTemplate(src, dst, nil); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (dst *Metal3MachineTemplate) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*v1alpha5.Metal3MachineTemplate)
+	if err := Convert_v1alpha5_Metal3MachineTemplate_To_v1alpha4_Metal3MachineTemplate(src, dst, nil); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (src *Metal3MachineTemplateList) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha5.Metal3MachineTemplateList)
+	return Convert_v1alpha4_Metal3MachineTemplateList_To_v1alpha5_Metal3MachineTemplateList(src, dst, nil)
+}
+
+func (dst *Metal3MachineTemplateList) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*v1alpha5.Metal3MachineTemplateList)
+	return Convert_v1alpha5_Metal3MachineTemplateList_To_v1alpha4_Metal3MachineTemplateList(src, dst, nil)
+}
```

Comment:
you're missing the Metal3Data, Metal3DataList, Metal3DataTemplate, Metal3DataTemplateList, Metal3DataClaim, Metal3DataClaimList ConvertFrom and ConvertTo functions

Reply:
Added